tekton/resources/nightly-tests/bastion-z/deploy_tekton_component.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: deploy-tekton-project-nightly
+spec:
+  params:
+  - name: package
+    description: package to install
+  - name: container-registry
+    description: container registry used to publish build images
+  - name: target-arch
+    description: target architecture for tests (s390x, ppc64le, arm64)
+  resources:
+    inputs:
+    - name: tekton-project-source
+      type: git
+      targetPath: src/$(params.package)
+  workspaces:
+  - name: k8s-shared
+    description: workspace for k8s config, configuration file is expected to have `config` name
+    mountPath: /root/.kube
+  steps:
+  - name: deploy
+    workingdir: /workspace/src/$(params.package)
+    image: gcr.io/tekton-releases/dogfooding/test-runner:latest
+    env:
+    - name: GOPATH
+      value: /workspace
+    - name: KO_DOCKER_REPO
+      value: $(params.container-registry)
+    - name: KUBECONFIG
+      value: $(workspaces.k8s-shared.path)/config
+    command:
+    - /bin/bash
+    args:
+    - -ce
+    - |
+      ko apply --platform=linux/$(params.target-arch) -f config/
```

Comment:
We could consider here installing our nightly builds instead, so we don't need to rebuild and we test our builds as well. We already have a task to deploy a pre-built tekton, we could then use that too.
Not something for this PR, but we could consider changing it as follow up.

Reply:
[e2e pipeline test ](https://github.com/tektoncd/pipeline/blob/f59e1bec5c6b6566370a5dd5f35671a49f577524/test/taskrun_test.go#L125) fails with nightly release (it checks the annotations and expects different value, then nightly release has)

+ testing the latest code instead of run the tests for the same release several times (in case of nightly release delay, for instance) looks better to me, however I am ok with nightly release approach for the future